spec/acceptance/progress_events_spec.rb

```diff
@@ -0,0 +1,918 @@
+  #
+  # After this, the dataset_items and progress events are:
+  # - 1 dataset item created by writer_user (create_entire_hierarchy)
+  #   - with 1 progress event created by no_access_user
+  # - 1 dataset item in the default dataset, created by writer_user
+  #   - with 1 progress event created by admin_user
+  # - 1 dataset item created by create_no_access_hierarchy
+  #   - with 1 progress event, created by a different user
+  # -> 3 progress events total
+
+  create_no_access_hierarchy
+
+  ################################
+  # INDEX
+  ################################
+
+  # users have read access on progress events if they have read
+  # access on the project via project/site/audio_recording/dataset_item/progress_event
+  # or if they are the creator
+
+  get '/progress_events' do
+    let(:authentication_token) { admin_token }
+    standard_request_options(
+        :get,
+        'INDEX (as admin)',
+        :ok,
+        {expected_json_path: 'data/0/dataset_item_id', data_item_count: 3}
+    )
+  end
+
+  # permissions for owner, writer and reader are the same as each other
+  non_admin_opts = {expected_json_path: 'data/0/dataset_item_id', data_item_count: 2}
+
+  get '/progress_events' do
+    let(:authentication_token) { owner_token }
+    standard_request_options(:get,'INDEX (as owner)',:ok, non_admin_opts)
+  end
+
+  get '/progress_events' do
+    let(:authentication_token) { writer_token }
+    standard_request_options(:get,'INDEX (as writer)',:ok, non_admin_opts)
+  end
+
+  get '/progress_events' do
+    let(:authentication_token) { reader_token }
+    standard_request_options(:get,'INDEX (as reader)',:ok, non_admin_opts)
+  end
+
+  get '/progress_events' do
+    let(:authentication_token) { no_access_token }
+    standard_request_options(
+        :get,
+        'INDEX (as no access user)',
+        :ok,
+        {response_body_content: ['"order_by":"created_at","direction":"desc"'], data_item_count: 0}
+    )
+  end
+
+  get '/progress_events' do
+    let(:authentication_token) { invalid_token }
+    standard_request_options(
+        :get,
+        'INDEX (invalid token)',
+        :unauthorized,
+        {expected_json_path: get_json_error_path(:sign_up)}
+    )
+  end
+
+  get '/progress_events' do
+    standard_request_options(
+        :get,
+        'INDEX (as anonymous user)',
+        :ok,
+        {remove_auth: true, response_body_content: ['"order_by":"created_at","direction":"desc"'], data_item_count: 0}
+    )
+  end
+
+  get '/progress_events' do
+    let(:authentication_token) { harvester_token }
+    standard_request_options(
+        :get,
+        'INDEX (as harvester)',
+        :forbidden,
+        {response_body_content: ['"data":null'], expected_json_path: get_json_error_path(:permissions)}
+    )
+  end
+
+
+  ################################
+  # CREATE
+  ################################
+
+  # any user can create a dataset item if they have read-access on the project
```

Comment:
~~this comment seems erroneous?~~

~~Only for the default dataset right?~~ 

Oh, I see, this is a copy-pasta comment. Please update.